File: rb/features.rb
# OsuBeatmap SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module OsuBeatmapFeatures
  def self.make_feature(name)
    case name
    when "base"
      OsuBeatmapBaseFeature.new
    when "test"
      OsuBeatmapTestFeature.new
    else
      OsuBeatmapBaseFeature.new
    end
  end
end

File: py/osubeatmap_sdk/features.py
# OsuBeatmap SDK feature factory

from osubeatmap_sdk.feature.base_feature import OsuBeatmapBaseFeature
from osubeatmap_sdk.feature.test_feature import OsuBeatmapTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OsuBeatmapBaseFeature(),
        "test": lambda: OsuBeatmapTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
